### PR TITLE
[doc]user defined var

### PR DESCRIPTION
**Pull request by Amber1990Zhang**
- Add user-defined variable doc.
- Update doc TOC.

### CODE REVIEW

File: docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# 用户自定义变量
+
+**Nebula Graph** 支持用户自定义变量，其格式为 `$ * var_name *`。其中变量名`* var_name *`由字母字符组成。目前不建议使用任何其他字符。用户自定义变量可暂时将值存储在前一个语句中的自定义变量中，并在另一个语句中使用，从而可将值从一个语句传递到另一个语句。
```
Comment (laura-ding) [inline docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md]: `$ * var_name *` and `* var_name *` remove the `*`

File: docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+用户定义变量只能在一次执行中使用（由分号（;）或管道（|）隔开的复合语句，并提交给服务器一起执行）。
+
+请注意，自定义的变量仅可在指定会话中执行。在一个语句中定义的自定义变量不能在其他客户端和其他执行中使用，也就是说定义语句和使用它的语句应该一起提交。当客户端发送执行时，变量会自动释放。
```
Comment (laura-ding) [inline docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md]: `自定义的变量仅可在指定会话中执行` 修改为 `自定义的变量仅在当前会话的当前query有效`

File: docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+用户变量名称区分大小写。建议名称的最大长度为256个字符。
```
Comment (laura-ding) [inline docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md]: 把`建议名称的最大长度为256个字符`删除了吧

File: docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+使用自定义变量的一种方法是使用 YIELD 语句：
+
+```ngql
+YIELD $var1 = expr [, $var2 = expr]
```
Comment (laura-ding) [inline docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md]: 这个例子没有实际意义而且语法也是错的，举一个有意义的，比如

```ngql
$var = GO FROM hash('Tom') OVER like YIELD like._dst AS id; GO FROM $var.id OVER study;
```

File: docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# User-Defined Variables
+
+**Nebula Graph** supports the concept of user-defined variables, which allows passing of a value from one statement to another. A user-defined variable in written as `$ var_name`, `var_name` is the name of variable and can consist of alphanumeric characters, any other characters are not recommended currently.
```
Comment (wilsonyou) [inline docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md]: **Nebula Graph** supports user-defined variables, which allows passing the result of one statement to another.

File: docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+User-defined variables can only be used in one execution (compound statements separated by semicolon (;) or pipe (|) and be submitted to server to execute together).
```
Comment (wilsonyou) [inline docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md]: User-defined variables can only be used in one execution (compound statements separated by semicolon (;) or pipe (|) and is submitted to the server to execute together).

File: docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+Be noted that user-defined variables are session and execution specific. A user-defined variable in one statement can NOT be used in neither other clients nor other executions, which means that the definition statement and the statements that use it should be submitted together. And when the session ends these variables are automatically expired.
```
Comment (wilsonyou) [inline docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md]: Be noted that user-defined variables are session and execution specific. -> 

Be noted that a user-defined variable is valid only at the current session and execution.

Comment (wilsonyou) [inline docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md]: A user-defined variable in one statement can NOT be used in other clients and executions, which means that the variable definition statement and the statements that use it must be submitted together. When the client sends the execution, the variable is automatically released.

File: docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# User-Defined Variables
+
+**Nebula Graph** supports user-defined variables, which allows passing the result of one statement to another. A user-defined variable in written as `$ var_name`, `var_name` is the name of variable and can consist of alphanumeric characters, any other characters are not recommended currently.
```
Comment (whitewum) [inline docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md]: `$ var_name` extra space
where `var_name` is a user-defined name/variable
(delete can) consist of alphanumeric characters

File: docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# 用户自定义变量
+
+**Nebula Graph** 支持用户自定义变量，其格式为 `$ var_name`。其中变量名 `var_name` 由字母字符组成。目前不建议使用任何其他字符。用户自定义变量可暂时将值存储在前一个语句中的自定义变量中，并在另一个语句中使用，从而可将值从一个语句传递到另一个语句。
+
+用户定义变量只能在一次执行中使用（由分号 `;` 或管道 `|` 隔开的复合语句，并提交给服务器一起执行）。
```
Comment (whitewum) [inline docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md]: 用户定义变量只能在一次执行中使用（由分号 `;` 或管道 `|` 隔开的复合语句)，并提交给服务器一起执行

File: docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
+# 用户自定义变量
+
+**Nebula Graph** 支持用户自定义变量，其格式为 `$ var_name`。其中变量名 `var_name` 由字母字符组成。目前不建议使用任何其他字符。用户自定义变量可暂时将值存储在前一个语句中的自定义变量中，并在另一个语句中使用，从而可将值从一个语句传递到另一个语句。
```
Comment (whitewum) [inline docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md]: 并在(稍后的)另一个语句中使用
从而可将中间结果从一个语句传递到另一个语句

File: docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+用户定义变量只能在一次执行中使用（由分号 `;` 或管道 `|` 隔开的复合语句，并提交给服务器一起执行）。
+
+请注意，自定义的变量仅在当前会话的当前查询有效。在一个语句中定义的自定义变量不能在其他客户端和其他执行中使用，也就是说定义语句和使用它的语句应该一起提交。当客户端发送执行时，变量会自动释放。
+
+用户变量名称区分大小写。
+
+使用自定义变量的一种方法是使用 YIELD 语句：
```
Comment (whitewum) [inline docs/manual-CN/2.query-language/3.language-structure/user-defined-variables.md]: 这句话没用了吧

File: docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md [hunk 12]
```diff
@@ -1,14 +1,14 @@
 # User-Defined Variables
 
-**Nebula Graph** supports user-defined variables, which allows passing the result of one statement to another. A user-defined variable in written as `$ var_name`, `var_name` is the name of variable and can consist of alphanumeric characters, any other characters are not recommended currently.
+**Nebula Graph** supports user-defined variables, which allows passing the result of one statement to another. A user-defined variable in written as `$var_name`, where `var_name` is a user-defined name/variable that consists of alphanumeric characters, any other characters are not recommended currently.
```
Comment (wilsonyou) [inline docs/manual-EN/2.query-language/3.language-structure/user-defined-variables.md]: is written

**Pull request merged**